cassiopeia/plotting/itol_utilities.py

```diff
@@ -0,0 +1,623 @@
+    outdf["cellBC"] = _leaves
+    outdf["gradient"] = df.values
+
+    header = [
+        "DATASET_GRADIENT",
+        "SEPARATOR TAB",
+        "COLOR\t#00000",
+        f"COLOR_MIN\t{color_min}",
+        f"COLOR_MAX\t{color_max}",
+        "MARGIN\t100",
+        f"DATASET_LABEL\t{df.name}",
+        "STRIP_WIDTH\t50",
+        "SHOW_INTERNAL\t0",
+        "DATA",
+        "",
+    ]
+
+    outfp = os.path.join(output_directory, f"{dataset_name}.{df.name}.txt")
+    with open(outfp, "w") as fOut:
+        for line in header:
+            fOut.write(line + "\n")
+        df_writeout = outdf.to_csv(
+            None, sep="\t", header=False, index=False
+        )
+        fOut.write(df_writeout)
+    return outfp
+
+
+def create_colorbar(
+    labels: pd.DataFrame,
+    tree: CassiopeiaTree,
+    colormap: Dict[str, Tuple[int, int, int]],
+    dataset_name: str,
+    output_directory: str = ".tmp/",
+    create_legend: bool = False,
+) -> str:
+    """Creates a colorbar file for the iTOL batch uploader
+
+    Creates a colorbar file for iTOL from categorical data. This will write out
+    the file to the specified location, which can then be uploaded to iTOL.
+
+    Args:
+        labels: A pandas series with categorical data (can be represented as strings
+            or categories)
+        tree: CassiopeiaTree
+        colormap: A mapping from category to RGB colors
+        dataset_name: Name for the dataset
+        output_directory: Where to write the output file
+        create_legend: Include legend for this colorbar.
+
+    Returns:
+        The filepath to new colorbar file.
+    """
+
+    _leaves = tree.leaves
+    labelcolors_iTOL = []
+    for i in labels.loc[_leaves].values:
+        colors_i = colormap[i]
+        color_i = (
+            "rgb("
+            + str(colors_i[0])
+            + ","
+            + str(colors_i[1])
+            + ","
+            + str(colors_i[2])
+            + ")"
+        )
+        labelcolors_iTOL.append(color_i)
+    dfCellColor = pd.DataFrame()
+    dfCellColor["cellBC"] = _leaves
+    dfCellColor["color"] = labelcolors_iTOL
+
+    # save file with header
+    header = [
+        "DATASET_COLORSTRIP",
+        "SEPARATOR TAB",
+        "COLOR\t#FF0000",
+        "MARGIN\t100",
+        f"DATASET_LABEL\t{dataset_name}",
+        "STRIP_WIDTH\t100",
+        "SHOW_INTERNAL\t0",
+        "",
+    ]
+
+    outfp = os.path.join(output_directory, f"{dataset_name}.txt")
+    with open(outfp, "w") as SIDout:
+        for line in header:
+            SIDout.write(line + "\n")
+
+        if create_legend:
+            number_of_items = len(colormap)
+
+            SIDout.write(f"LEGEND_TITLE\t{dataset_name} legend\n")
+            SIDout.write("LEGEND_SHAPES")
+            for _ in range(number_of_items):
+                SIDout.write("\t1")
+
+            SIDout.write("\nLEGEND_COLORS")
+            for col in colormap.values():
+                SIDout.write(f"\t{rgb_to_hex(col)}")
```

Comment:
Where is `rgb_to_hex` defined?

Reply:
Great catch.. I hadn't defined it! Goes to show why code coverage is so important...